src/main/java/org/betonquest/betonquest/utils/Utils.java

```diff
@@ -166,20 +167,38 @@ public static boolean isQuestItem(final ItemStack item) {
                 && item.getItemMeta().getLore().contains(Config.getMessage(Config.getLanguage(), "quest_item"));
     }
 
+    /**
+     * Gets the party of the player.
+     * A range of 0 means worldwide and -1 means server-wide.
+     *
+     * @param onlineProfile the player to get the party of
+     * @param range         the range of the party
+     * @param conditions    conditions that the party members must meet
+     * @return the party of the player
+     */
     @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
     public static Map<OnlineProfile, Double> getParty(final OnlineProfile onlineProfile, final double range, final ConditionID... conditions) {
         final Location loc = onlineProfile.getPlayer().getLocation();
         final World world = loc.getWorld();
         final double squared = range * range;
 
-        return PlayerConverter.getOnlineProfiles().stream()
-                .filter(profile -> world.equals(profile.getPlayer().getWorld()))
-                .map(profile -> Pair.of(profile, profile.getPlayer().getLocation().distanceSquared(loc)))
-                .filter(pair -> pair.right() <= squared)
+        final Stream<OnlineProfile> players = PlayerConverter.getOnlineProfiles().stream();
+        final Stream<OnlineProfile> worldPlayers = range == -1 ? players : players.filter(profile -> world.equals(profile.getPlayer().getWorld()));
+        final Stream<Pair<OnlineProfile, Double>> distancePlayers = worldPlayers.map(profile -> Pair.of(profile, getDistanceSquared(profile, loc)));
+        final Stream<Pair<OnlineProfile, Double>> rangePlayers = range <= 0 ? distancePlayers : distancePlayers.filter(pair -> pair.right() <= squared);
+        return rangePlayers
```

Comment:
Your solution is fine but I'dd suggest the following alternative as I personally find it more readable:

```suggestion
        return PlayerConverter.getOnlineProfiles().stream()
                .filter(profile -> range == -1 || world.equals(profile.getPlayer().getWorld()))
                .map(profile -> Pair.of(profile, getDistanceSquared(profile, loc)))
                .filter(pair -> range <= 0 || pair.right() <= squared)
```

Decide whatever you prefer and either add the suggestion or just merge.